src/libmongoc/tests/bsonutil/bson-match.c

```diff
@@ -498,7 +500,7 @@ bson_matcher_match (bson_matcher_t *matcher,
                                   expected_val,
                                   actual_val,
                                   path_child,
-                                  allow_extra,
+                                  (is_root && array_of_root_docs),
```

Comment:
Sanity check appreciated. I believe with this logic `array_of_root_docs` only gets inherited when we recurse from handling a root BSON array. And per the code above, the we ignore extra keys if the root is either a document or a child document of a root array.

Reply:
Without `array_of_root_docs` coming from `bson_matcher_match`, we might consider _any_ array of BSON values to be an array of docs. Consider the case where we're matching an array of embedded documents returned by `distinct`. Those should _not_ allow extra fields within.